nengo/connection.py

```diff
@@ -122,20 +123,28 @@ class ConnectionSolverParam(SolverParam):
     def coerce(self, conn, solver):
         solver = super().coerce(conn, solver)
         if solver is not None:
-            if solver.weights and not isinstance(conn.pre_obj, Ensemble):
+            if solver.weights and not isinstance(conn.pre_obj, (Ensemble, Node)):
                 raise ValidationError(
                     "weight solvers only work for connections from ensembles "
                     "(got %r)" % type(conn.pre_obj).__name__,
                     attr=self.name,
                     obj=conn,
                 )
-            if solver.weights and not isinstance(conn.post_obj, Ensemble):
+            elif solver.weights and isinstance(conn.pre_obj, Node):
+                warnings.warn(
+                    "weight solvers only work for connections to ensembles, not nodes"
```

Comment:
Should say "from ensembles", not "to ensembles".

I'm actually wondering whether we want a warning at all in this case. For Node->Ensemble connections, using a solver with `weights=True` will result in weights that go from the node directly to the neurons (i.e. the encoders are incorporated into the weights). This is different from one with `weights=False` (where weights and encoders are separate), so there is a reason that someone might want to set `weights=True` on a Node->Ensemble connection.